OsmAnd/src/net/osmand/plus/settings/backend/OsmandSettings.java

```diff
@@ -446,22 +448,57 @@ public boolean switchAppModeToPrevious() {
 
 	public boolean switchAppMode(boolean next) {
 		ApplicationMode appMode = getApplicationMode();
+		ApplicationMode nextAppMode = getSwitchedAppMode(appMode, next);
+		if (appMode != nextAppMode && setApplicationMode(nextAppMode)) {
+			String pattern = ctx.getString(R.string.application_profile_changed);
+			String message = String.format(pattern, nextAppMode.toHumanString());
+			ctx.showShortToastMessage(message);
+			return true;
+		}
+		return false;
+	}
+
+	public ApplicationMode getSwitchedAppMode(ApplicationMode selectedMode, boolean next) {
 		List<ApplicationMode> enabledModes = ApplicationMode.values(ctx);
-		int indexOfCurrent = enabledModes.indexOf(appMode);
+		int indexOfCurrent = enabledModes.indexOf(selectedMode);
 		int indexOfNext;
 		if (next) {
 			indexOfNext = indexOfCurrent < enabledModes.size() - 1 ? indexOfCurrent + 1 : 0;
 		} else {
 			indexOfNext = indexOfCurrent > 0 ? indexOfCurrent - 1 : enabledModes.size() - 1;
 		}
-		ApplicationMode nextAppMode = enabledModes.get(indexOfNext);
-		if (appMode != nextAppMode && setApplicationMode(nextAppMode)) {
-			String pattern = ctx.getString(R.string.application_profile_changed);
-			String message = String.format(pattern, nextAppMode.toHumanString());
-			ctx.showShortToastMessage(message);
-			return true;
+		return enabledModes.get(indexOfNext);
+	}
+
+	private final Handler delayedSwitchProfileHandler = new Handler();
+	private ApplicationMode delayedSwitchProfile;
+	private Toast delayedSwitchProfileToast;
+
+	public void delayedSwitchAppMode(boolean next) {
```

Comment:
Probably better to store this behaviour in quick action, because we need it only there